ethpm_types/contract_type.py

```diff
@@ -324,9 +324,34 @@ def get_deployment_bytecode(self) -> Optional[HexBytes]:
 
         return None
 
+    @cached_property
+    def selector_identifiers(self) -> Dict[str, str]:
+        """
+        Returns a mapping of the full suite of signatures to selectors/topics/IDs for this
+        contract.
+        """
+
+        def get_id(aitem: ABI) -> str:
+            if isinstance(aitem, MethodABI) or isinstance(aitem, ErrorABI):
+                return HexBytes(self._selector_hash_fn(aitem.selector)[:4]).hex()
+            else:
+                assert hasattr(aitem, "selector")
+                return HexBytes(self._selector_hash_fn(aitem.selector)).hex()
+
+        abis_with_selector = [x for x in self.abi if hasattr(x, "selector")]
+        return {x.selector: get_id(x) for x in abis_with_selector}
+
+    @cached_property
+    def identifier_lookup(self) -> Dict[str, str]:
```

Comment:
thought: @antazoey I'm kind of wondering if this should be `Dict[str, ABIType]` now.  Seems like it might be more useful to get the full ABI object rather than just the text sig/definition/whatever.

This kind of access makes a bit of sense to me;

```python
my_contract.identifier_lookup['0x27e235e3'].selector
```

It's less of a direct path to what I'm looking for but it might be more useful to others.

Reply:
That is cool. Do you think we should also add `selector_hash` to ABIType so users dont have to keccak themselves?

Reply:
oh wait the hash is the key, nevermind
ya this works